analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Invocation.cs

```diff
@@ -1260,6 +1255,53 @@ public void Invocation_BoolEquals_BothArgumentsHaveNoConstraint_DoesNotLearn()
         validator.TagValue("Result").Should().HaveOnlyConstraint(ObjectConstraint.NotNull);
     }
 
+    [DataTestMethod]
+    [DataRow("42", "42", true)]
+    [DataRow("0", "42", false)]
+    [DataRow("((int?)42)", "((int?)42)", true)]
+    [DataRow("((int?)42)", "((int?)0)", false)]
+    public void Invocation_NumberEquals_SingleValues_LearnsResult(string left, string right, bool expected)
+    {
+        var code = $"""
+                var result = {left}.Equals({right});
+                Tag("Result", result);
+
+                var resultStaticCall = object.Equals({left}, {right});
+                Tag("ResultStaticCall", resultStaticCall);
+                """;
+        var validator = SETestContext.CreateCS(code).Validator;
+        validator.TagValue("Result").Should().HaveOnlyConstraints(ObjectConstraint.NotNull, BoolConstraint.From(expected));
+        validator.TagValue("ResultStaticCall").Should().HaveOnlyConstraints(ObjectConstraint.NotNull, BoolConstraint.From(expected));
+    }
+
+    [TestMethod]
+    public void Invocation_NumberEquals_DoesNotLearn()
+    {
+        var code = """
+                if (i > 1 && i < 10)
+                {
+                    var result = object.Equals(i, 4);
+                    Tag("Result", result);
+                }
+                """;
+        var validator = SETestContext.CreateCS(code, "int i").Validator;
+        validator.TagValue("Result").Should().HaveOnlyConstraints(ObjectConstraint.NotNull); // Should learn 'false'
+    }
+
+    [TestMethod]
+    public void Invocation_NumberEquals_RangesDoNotOverlap_LearnsResult()
+    {
+        var code = $$"""
+                if (i > 0 && j < 0)
```

Comment:
Same thing, easier to imagine (negative first, on the left of the axis)
```suggestion
                if (i < 0 && j > 0)
```